### PR TITLE
add a "command list" to admin commands

### PR DESCRIPTION
**Pull request by KfreeZ**
hi, this is a little enhancement proposal of adding a  command list to admin commands.

1. add a command list entry point at / for admin commands:
> ```
> $ curl http://10.239.241.90:15021/
> /healthz/ready                  print server state
> /debug/pprof/profile            print the pprof data
> /debug/gprof/profile            print the gprof data
> /debug/gprof/heap               print the heap data
> /quitquitquit                   exit the server
> /config_dump                    print the workload information
> /                               list all the commands
> ```

2. if a URL is invalid, hint the user can use / to get the command list:
> ```
> $ curl http://10.239.241.90:15021/wrong
> Invalid command.
> Please goto / to get the command list
> ```

3. regression tests 
>```
> $ curl http://10.239.241.90:15021/healthz/ready
> ready
> 
> $ curl http://10.239.241.90:15021/debug/pprof/profile
> 
> _c
> |��
> 
> 
> A��=�S�
> 
> 
> A��=�""""      "�" "�" "�" "�"  �"      "      �"
> "
> e"
>        "
>         �"
>                "
> "E"   "�"D"�"  "�" "�"["     "�"""""["     "�"|"     "�""      "�"  " �" !"!�"      """�"#"#"     $"$�" %"%�" &"&�'"'("(=" )")�" *"*�" +"+�" ,",�-"-.".D" /"/�" 0"0�" 1"1�2"2="     3"3�"      4"4�5"5*"     6"6�" 7"7�" 8"8�" 9"9�:":z"     ;";�" <"<�" ="=�" >">?"?l"      @"@�A"A_"     B"B�    "  C"C�*  u� *       � & 00 kk  IILL *pp <*
>  [*
>  tt [*
>                ++ �*
> �� �*
> �� h*      �
>                     d*
> �� sJ 8*       9 �*       �n =*       a@ �z T*  W� !*       r4 �*       � ,*
>  �� Z A*       x �*
> �� D*
> Warning: Binary output can mess up your terminal. Use "--output -" to tell
> Warning: curl to output it to your terminal anyway, or consider "--output
> Warning: <FILE>" to save to a file.
> V�
> 
> $ curl http://10.239.241.90:15021/debug/gprof/profile
> gperftools not enabled
> 
> $ curl http://10.239.241.90:15021/debug/gprof/heap
> gperftools heap not enable
> 
> $ curl http://10.239.241.90:15021/config_dump
> {"workloads":{},"workload_to_vip":{},"vips":{}
> 
> $ curl http://10.239.241.90:15021/wrong
> Invalid command.
> Please goto / to get the command list
> 
> zkf@pinn:~$ curl -X POST http://10.239.241.90:15021/quitquitquit
> shutdown now
>```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
Would expect access http://xxxxxx:15021/ could return such info automatically instead of hard coding

File: src/admin/mod.rs [hunk 1]
```diff
@@ -287,6 +288,13 @@ async fn handle_config_dump(dump: WorkloadInformation, _req: Request<Body>) -> R
         .unwrap()
 }
 
+async fn handle_help(_req: Request<Body>) -> Response<Body> {
+    Response::builder()
+        .status(hyper::StatusCode::OK)
+        .body("/healths/ready\t\t\tprint server state\n/debug/pprof/profile\t\tprint the profile data\n/debug/gprof/profile\t\tprint the gprof\n/debug/gprof/heap\t\tprint the heap profile\n/quitquitquit\t\t\texit the ztunnel\n/config_dump\t\t\tprint the workload information\n/help\t\t\t\tprint the command list\n".into())
```
Comment (PiotrSikora) [inline src/admin/mod.rs:294]: Could you convert this to a readable multi-line string?

File: src/admin/mod.rs [hunk 2]
```diff
@@ -180,10 +180,11 @@ impl Server {
                                 "/config_dump" => Ok::<_, hyper::Error>(
                                     handle_config_dump(workload_info, req).await,
                                 ),
+                                "/help" => Ok::<_, hyper::Error>(handle_help(req).await),
```
Comment (PiotrSikora) [inline src/admin/mod.rs:183]: I agree with @hzxuzhonghu, this should be served as a "menu" when visting `/`.

Comment (KfreeZ) [reply src/admin/mod.rs:183]: > 

make sense, thanks you @hzxuzhonghu @PiotrSikora, will change it

File: src/admin/mod.rs (hunk 1 above)
Comment (KfreeZ) [reply src/admin/mod.rs:294]: > Could you convert this to a readable multi-line string?
I am working on it, but my limited rust skill make it looks ugly, will send it out when I fix it.

File: src/admin/mod.rs (hunk 2 above)
Comment (KfreeZ) [reply src/admin/mod.rs]: @hzxuzhonghu @PiotrSikora 
I updated the code, alos put some information and logs in the PR comments. please have a look. Thank you.

**Comment by KfreeZ** [comment]
hello, @howardjohn, I got some issue when I rebased this branch and accidently deleted it, so I closed this PR.
I'd like to know if you think this PR is valuable, if not I will ignore it, if yes, I will re-open a new PR for reviewing.
you can see the PR log for the details and all the code change is in the admin/mod.rs and please ignore others.

**Pull request closed without merging**